### ISSUE TITLE
[release/2.12] PT2E INT8 accuracy regression on shufflenet_v2_x1_0 (XPU): Acc@1 66.5 → 0.09 between release/2.11 and 2.12.0+xpu

### ISSUE THREAD

**Issue by mengfei25** [intel/torch-xpu-ops#3489]
### Summary
On `torch==2.12.0+xpu`, PT2E (inductor) INT8 quantization of `shufflenet_v2_x1_0` on XPU collapses to near-random accuracy. The same flow on `release/2.11` reaches Acc@1 66.5 / Acc@5 86.528, and FP32 on the same 2.12.0+xpu build is healthy. This is a regression in the XPU PT2E INT8 path.

### Environment
- `torch`: `2.12.0+xpu`
- Python 3.12 (venv at `/workspace/myvenv`)
- Device: XPU
- Quant config: default, PT2E flow via `pt2e-accuracy/scripts/modelbench/quant/inductor_quant_acc.py`
- Dataset: ImageNet ILSVRC2012 val (standard valprep)

### Reproduce
```bash

git clone -b main https://github.com/chuanqi129/inductor-tools pt2e-accuracy

ACC_CMD="python pt2e-accuracy/scripts/modelbench/quant/inductor_quant_acc.py \
         --device xpu --dataset_dir ${DATASET_DIR}"
m=shufflenet_v2_x1_0

# INT8 (broken on 2.12.0+xpu)
${ACC_CMD} --model_list "${m}"

# FP32 (healthy, sanity check)
${ACC_CMD} --model_list "${m}" --is_fp32
```

### Results

| Build | Mode | Acc@1 | Acc@5 |
|---|---|---|---|
| release/2.11 | INT8 (default config) | **66.500** | **86.528** |
| 2.12.0+xpu | INT8 (default config) | **0.090** | **0.482** |
| 2.12.0+xpu | FP32 | 69.356 | 88.314 |

### Observed log (INT8 run, 2.12.0+xpu, abridged)
```
using ptq
W0427 05:36:24 torch/_inductor/codecache.py:832] [1/0] pre_grad_custom_pass function does not implement uuid(); falling back to early timing ...
/workspace/myvenv/lib/python3.12/site-packages/torch/_inductor/mkldnn_lowerings.py:763: UserWarning:
  To copy construct from a tensor, it is recommended to use sourceTensor.detach().clone() ...
  torch.tensor(w_zp_tensor, dtype=torch.int32), name=w_zp.get_name()
W0427 05:36:28 torch/_inductor/select_algorithm.py:3254] [1/0] Constructing input/output tensor meta failed for Extern Choice

shufflenet_v2_x1_0 int8:  * Acc@1 0.090 Acc@5 0.482
```

(The two `torchao` `_C_mxfp8.so` / `_C_cutlass_90a.abi3.so` "Failed to load" lines at startup are unrelated CUDA-only artifacts shipped in the wheel.)

### Analysis (PyTorch + torch-xpu-ops code refs)

The two warnings emitted only on the INT8 path point at the inductor PT2E lowering for XPU, which is shared with CPU via the mkldnn lowerings:

1. **Weight zero-point handling** — `torch/_inductor/mkldnn_lowerings.py` around L763, in the QConv/QLinear PT2E lowering:
   ```python
   w_zp_tensor = V.graph.constants[w_zp.get_name()].to(torch.int32)
   w_zp = V.graph.add_tensor_constant(
       torch.tensor(w_zp_tensor, dtype=torch.int32),  # <-- triggers the warning
       name=w_zp.get_name(),
   )
   ```
   The `torch.tensor(<existing tensor>, dtype=int32)` re-wrap is what produces the user-warning, and is the canonical path for `w_zp` on the PT2E flow. A dtype/layout mismatch here would directly corrupt weight quantization math.

2. **Extern-kernel meta construction failure** — `torch/_inductor/select_algorithm.py` around L3254:
   ```python
   try:
       self.input_tensor_meta  = TensorMeta.from_irnodes(self.input_nodes)
       self.output_tensor_meta = TensorMeta.from_irnodes(self.layout)
   except Exception:
       log.warning("Constructing input/output tensor meta failed for Extern Choice")
   ```
   On XPU this exception is being swallowed for the quantized extern choice, so algorithm selection proceeds with empty meta and an extern fallback may be picked / dispatched with the wrong layout. This warning does **not** appear on the FP32 run.

3. **`QConvPointWisePT2E` is shared CPU/XPU** — `torch/_inductor/mkldnn_ir.py` (`QConvPointWisePT2E.create` → `_prepare_convolution_fusion_create`, gated by `SUPPORTED_MKLDNN_DEVICES = ("cpu", "xpu")` in `torch/_inductor/utils.py`). The XPU side delegates to oneDNN graph via the kernel name `aoti_torch_xpu__qconv_pointwise_tensor`. Any change in the shared lowering between 2.11 and 2.12 (w_zp dtype/stride, channels-last propagation, grouped-conv handling) would land on XPU even if originally targeted at CPU.

`shufflenet_v2_x1_0` is depthwise/grouped-conv heavy and uses channel-shuffle, which historically has been the most fragile shape for INT8 conv lowerings — consistent with this being the first model to fail.

### Likely root-cause hypotheses (ranked)
1. **Weight zero-point dtype/layout regression** in the shared mkldnn PT2E lowering (`mkldnn_lowerings.py` near L763) — the warning is emitted on every INT8 run.
2. **Extern-choice TensorMeta failure on XPU** (`select_algorithm.py` L3254) leading to wrong kernel selection for `qconv_pointwise` on XPU.
3. **Grouped/depthwise INT8 lowering gap on XPU** in `_prepare_convolution_fusion_create` (`mkldnn_ir.py`) — would explain why shufflenet is hit first.

**Comment by chunhuanMeng** [comment]
Tested on `torch 2.13.0.dev20260512+xpu` with Intel Data Center GPU Max 1550. The INT8 accuracy regression is **no longer reproducible** on 2.13:

| Config | Acc@1 | Acc@5 |
|--------|-------|-------|
| FP32 (2.13) | 69.356 | 88.314 |
| INT8 (2.13) | 66.364 | 86.480 |

The INT8 accuracy (66.364) is back to the expected range, matching the release/2.11 baseline. Please double-check.

**Comment by kaileiyx** [comment]
Verified with torch 2.13.0.dev20260520+xpu. The INT8 accuracy regression observed in 2.13 can no longer be reproduced.

shufflenet_v2_x1_0 int8:  * Acc@1  66.440    Acc@5  86.546
shufflenet_v2_x1_0 fp32:  * Acc@1  69.356   Acc@5  88.314

**Comment by chuanqi129** [comment]
Close it because it can't reproduce on main
